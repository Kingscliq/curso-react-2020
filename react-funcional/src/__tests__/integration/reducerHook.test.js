import "@testing-library/jest-dom";
import { render , act} from "./../../../test-utils";
import "@testing-library/jest-dom/extend-expect";
import userEvent from "@testing-library/user-event";

import ReducerHook from "./../../reducer/reducerHook";
import React from "react";

let documentBody;

describe("Render correctly Reducer Hook", () => {
  beforeEach(() => {
    documentBody = render(<ReducerHook />);
  });
  it("should show the title of system", () => {
    const { getByText } = documentBody;
    const title = getByText("Calculadora Simples");
    expect(title).toBeInTheDocument();
  });

  it("should show the Subtrair label in calc", () => {
    const { getByText } = documentBody;
    const subtrairLabel = getByText(/Subtrair/i);
    expect(subtrairLabel).toBeInTheDocument();
  });

  it("should show the Somar label in calc", () => {
    const { getByText } = documentBody;
    const somarLabel = getByText("Somar");
    expect(somarLabel).toBeInTheDocument();
  });

  it("should show the Resultado label in calc", () => {
    const { getByText } = documentBody;
    const resultadoLabel = getByText("Resultado:");
    expect(resultadoLabel).toBeInTheDocument();
  });

  it("should input number in first field", () => {
    const { getByTestId } = documentBody;
    const inputFirstNumber = getByTestId("firstInputCalc");
    expect(userEvent.type(inputFirstNumber, "11")).toBeTruthy();
  });

  it("should input number in second field", () => {
    const { getByTestId } = documentBody;
    const inputSecondNumber = getByTestId("secondInputCalc");
    expect(userEvent.type(inputSecondNumber, "9")).toBeTruthy();
  });

  it("should click in sum button", () => {
    const { getByRole } = documentBody;
    const btn = getByRole('button', {
      name: /Somar/i
    })
    act(() => {
      userEvent.click(btn);
    });
  });

  it("should click in subtract button", () => {
    const { getByRole } = documentBody;
    const btn = getByRole('button', {
      name: /Subtrair/i
    })
    act(() => {
      userEvent.click(btn);
    });
  });

  it("should be generate snapshot", () => {
    const { container } = documentBody;
    expect(container).toMatchSnapshot();
  });
});
