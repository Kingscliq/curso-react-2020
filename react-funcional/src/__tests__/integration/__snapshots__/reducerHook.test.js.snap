// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly Reducer Hook should be generate snapshot 1`] = `
<div>
  <div
    class="sc-dlfnuX bwzoMI"
  >
    <h1
      class="sc-hKgJUU hfvrQH"
    >
      Calculadora Simples
    </h1>
    Número:
    <br />
    <input
      class="ant-input sc-gsTEea ktcgBc"
      data-testid="firstInputCalc"
      type="text"
      value=""
    />
    <br />
    Número:
    <br />
    <input
      class="ant-input sc-gsTEea ktcgBc"
      data-testid="secondInputCalc"
      type="text"
      value=""
    />
    <br />
    <div
      class="sc-jSgvzq lnKEnR"
    >
      <button
        class="ant-btn ant-btn-primary sc-bdfBQB kcuiSJ"
        type="button"
      >
        <span>
          Somar
        </span>
      </button>
      <button
        class="ant-btn ant-btn-primary ant-btn-dangerous sc-bdfBQB kcuiSJ"
        type="button"
      >
        <span>
          Subtrair
        </span>
      </button>
    </div>
    <br />
    <div
      class="sc-eCstlR bjwjzI"
    >
      Resultado:
      <br />
      <input
        class="ant-input sc-gsTEea ktcgBc"
        readonly=""
        type="text"
        value=""
      />
    </div>
    <br />
  </div>
</div>
`;
